src/main/java/org/ibp/api/rest/role/RoleResource.java

```diff
@@ -30,4 +31,14 @@
 		return new ResponseEntity<>(this.roleService.getRoles(searchDTO), HttpStatus.OK);
 	}
 
+	@ApiOperation(value = "Save role", notes = "Save role. ")
+	@RequestMapping(value = "/roles", method = RequestMethod.POST)
+	@ResponseBody
+	public ResponseEntity createRole(@RequestBody final RoleGeneratorInput dto) {
+		Preconditions.checkNotNull(dto.getName(), "name cannot be null");
```

Comment:
@vmaletta Please do not use preconditions to validate the input. Move both to the role validator function